### ISSUE TITLE
Horizontal scroll just glitches on Windows

### ISSUE THREAD

**Issue by thomasmoors** [dbeaver/dbeaver#8404]
#### System information: 
Windows 10 Pro Versie 10.0.18363 Build 18363

- DBeaver version 7.0.2.202004051509
- Additional extensions

#### Connection specification:
- Database name and version
- Driver name MySql
- Do you use tunnels or proxies (SSH, SOCKS, etc)? 

#### Describe the problem you're observing:

When using a horizontal scrollwheel and the data extends the width of the screen, scrolling does not work using the mouse.
![image](https://user-images.githubusercontent.com/486765/78968080-20499400-7b04-11ea-88d2-ecf7288eb682.png)

#### Steps to reproduce, if exist:
Just scroll when viewing a table's data
#### Include any warning/errors/backtraces from the logs

![scroll](https://user-images.githubusercontent.com/486765/78968894-f7c29980-7b05-11ea-9782-e3290716feed.gif)

**Comment by kseniiaguzeeva** [comment]
Thank you for the bug report.

**Comment by serge-rider** [comment]
Fixed (hscroll handler added).
@kseniiaguzeeva Please test with real mouse scroller.

Workaround: do regular scroll (mouse wheel + press shift key).

**Comment by thomasmoors** [comment]
> Fixed (hscroll handler added).
> @kseniiaguzeeva Please test with real mouse scroller.
> 
> Workaround: do regular scroll (mouse wheel + press shift key).

It's fixed in the next version? Currently in 7.1.0 the problem still exists.
The workaround does work however!

**Comment by kseniiaguzeeva** [comment]
I can reproduce it with mouse Logitech MX Master 3.
But workaround works correctly.
(build   7.1.1.202006161955)

**Comment by serge-rider** [comment]
+1 fix

**Comment by kseniiaguzeeva** [comment]
verified.

**Comment by jaallen** [comment]
This issue is still happening in 7.3.2 with my MX Master 3

**Comment by jake-dev-99** [comment]
While the horizontal scrolling itself is now implemented, it is not currently working in 7.3.2. What usually happens is the horizontal scroll bar will flash repeatedly as it receives the scroll command, but it will not actually move. Occasionally, it will work as expected. I haven't found a pattern to this behavior yet as sometimes it happens with new queries, then fixes itself. Other times, I'll scroll correctly to one side, then it will start to act up. Other times, I will switch to another application and return, then it will be fixed. I'm using a MX Master 3 mouse.

Using the workaround works, but is not ideal for a production workday.

Here's a brief gif showing the behavior:

![dbeaver_horizontal_scroll](https://user-images.githubusercontent.com/6440039/103708147-df808d80-4f64-11eb-92f5-a49c34d20dc6.gif)

**Comment by kseniiaguzeeva** [comment]
Please tell also your OS.

**Comment by jake-dev-99** [comment]
Currently on Windows 10 2004, build 19041.572

**Comment by kseniiaguzeeva** [comment]
While I cannot reproduce the issue with the same mouse.

**Comment by kseniiaguzeeva** [comment]
Does horizontal scroll work in other applications? Do you have another eclipse based application to test it there?

**Comment by Gunshin82** [comment]
@kseniiaguzeeva, the horizontal scroll works fine in Excel, TextPad and Chrome. For SQL Server Management Studio, it works in the scripting part but not in the result pane. Had to change it from Horizontal Scroll to Zoom for it to work in the result. This workaround didn't work for DBeaver. In NotePad++, it doesn't some funky stuff with the scroll bar but doesn't scroll.

**Comment by jake-dev-99** [comment]
@kseniiaguzeeva Apologies for not responding. Unfortunately I was on vacation when you replied and it slipped through the cracks. I have similar behavior to @Gunshin82, although sometimes the editing pane also does not scroll correctly. Also Notepad++ works fine for me. Actually every other app I've tested has worked without any issues. However, I downloaded Eclipse itself and this is showing the same horizontal scrolling behavior as DBeaver, so it's likely related.

I also played around with various mouse settings in case it was related but this did not appear to make any difference.

**Comment by kseniiaguzeeva** [comment]
Thank you for the additional details. I will what we can do.

**Comment by Gunshin82** [comment]
@jaallen, I've just updated Notepad++ and it works perfectly. It's been awhile since I had last updated. Thanks for this ;-)

**Comment by kseniiaguzeeva** [comment]
Thank you for the report. Unfortunately we have to wait for Eclipse update.

**Comment by s2mako** [comment]
Are there any news on this issue? Unfortunately I'm also encountering this stuttering behaviour.

**Comment by parandoodudge** [comment]
Still encountering with my mx master 2. Just like @jaallen described.

**Comment by Gunshin82** [comment]
A work around has been supplied by alibarut on the Notepad++'s GitHub. Running the software as Administrator will allow the horizontal scrolling to work properly. I've tested this in both Notepad++ and DBeaver and they're working fine now. The downside is that you need Administrator privileges or have an Admin create a task which launches the software with highest privileges and then setup a shortcut which will run the task. Search the web for "Elevated Shortcut without UAC Prompt" for that second one.

**Comment by Pil0tXia** [comment]
I would be glad if this issue can be fixed. 😉

**Comment by Pil0tXia** [comment]
@arhayka Hi! Do you need any help to reproduce this bug? I've been coming to this page and following the progress many times. I hope this issue won't meet with too much resistance.

**Comment by arhayka** [comment]
Hi @Pil0tXia, thanks for your willingness to help! It would be great if you could confirm that the error is still reproducible. 

But we still need to reproduce it on our own to be able to fix it. We will try to do that in one of the next sprints.

**Comment by Pil0tXia** [comment]
@arhayka Yes, it can be reproduced steadily on 23.1.5.  I have the Logi MX Master 3S mouse and you may remote desktop my PC for debugging.

For now, two-finger left/right swiping is normal, but horizontal scrolling triggered by the secondary mouse wheel doesn't work in dbeaver. This is most likely due to the fact that dbeaver's two-finger scrolling is implemented in an insufficiently generic way, because for mice that can scroll horizontally, such as the M720/Master Series, the horizontal scrolling is implemented by the Logitech Options/Logi Options+ software, and it is well supported by most of the software.

Also, I had a very fortuitous encounter with the horizontal scrolling feature of the mouse working correctly, it happened to be triggered after I upgraded to 23.1.3 and performed a series of complex UI operations, the last one I remember was after executing a sql query that opened a small pane on the right side of the screen that displays some information about the database/query. Unfortunately, when I clicked on other areas of the dbeaver, the horizontal scrolling was once again unavailable and I was never able to reproduce the normal use situation.

**Comment by LonwoLonwo** [comment]
I can't reproduce issues with horizontal scrolling in the data viewer (or SQL Editor) with MX Anywhere 3 Logitech.

![Fn0PGO5gaf](https://github.com/dbeaver/dbeaver/assets/45152336/f032c6f7-768c-4d1c-bb03-65440e5cfb94)

We have other places in DBeaver - Eclipse controls - that must be fixed. 
But it works for the data viewer and in the result set viewer.

**Comment by Pil0tXia** [comment]
@LonwoLonwo Thank you for your assistance. Please be noticed that MX Master 2S/3/3S has a horizontal side rollwheel, however MX Anywhere 3 doesn't have one.

**Comment by LonwoLonwo** [comment]
For now, we do not have it for testing. 
We will wait for more reactions to this ticket to see how many people have this issue in the latest DBeaver versions.

**Comment by Pil0tXia** [comment]
@LonwoLonwo I don't think this issue is essentially discussing support for mouse hardware, but rather support for a common left/right swipe function call for the Windows platform. Because the left/right swipe function of these mice does not come from the hardware, but from the driver software. And the left/right swipe provided by the driver software works fine with other software (browsers, IDEs, etc.), but not with DBeaver.

For a detailed description and analysis of the driver software side, you may read this issue: https://github.com/dbeaver/dbeaver/issues/20518

**Comment by Pil0tXia** [comment]
@LonwoLonwo Hi, I've known that MX Anywhere 3 has an ability to swipe horizental with the side button (at thumb position )pressed down. It seems that it is the same with MX Master 3's, if the driver software these two are using are both Logitech Options or Logitech Options+.

**Comment by kirallam-viooh** [comment]
By any chance, on DBeaver > Preferences > Keys, I see commands for "Scroll Line Up" and "Scroll Line Down"... is it possible to add additional commands such as "Scroll Line Left" and "Scroll Line Right" such that I could configure the hotkeys there and add that onto MX Master 3s' Logi Option+ config?

**Comment by Pil0tXia** [comment]
@kirallam-viooh Possible but compromised. Good idea after all.

**Comment by GeniusBulldog** [comment]
> For now, we do not have it for testing. We will wait for more reactions to this ticket to see how many people have this issue in the latest DBeaver versions.

+1    have the same problem with my MX Master 3. Hope there is a way to solve it

**Comment by GeniusBulldog** [comment]
Maybe an important note: 

I have noticed that horizontal scrolling works with the MX3 in the editor field. Only in the results it does not work.

**Comment by MatiasToroH** [comment]
Any updates, i have the same issue but only on windows, Ubuntu 20.04 work fine besides it doesn't really having the logi+ options for the OS. it would be great if that function can work on windows

**Comment by CComparon** [comment]
I'm also unable to scroll horizontally with my [Logitech Lift Vertical Ergonomic Mouse](https://www.amazon.ca/gp/product/B09J1TB35S/ref=ppx_yo_dt_b_asin_title_o04_s00?ie=UTF8&th=1) on DBeaver 23.x on Windows 11 ☹️

**Comment by PaxonF** [comment]
Same issue here on Windows 10 with a Logitech MX Master 3 Mouse.

**Comment by turbo25** [comment]
The same issue for years now, using Logitech M705. I found a working scenario though.
If you use 2+ monitors, and move the focus to another window/screen, the horizontal scroll works as it should in the data viewer or any other if you move (without clicking) the mouse on them.
Ideally, it works too if you have an active window above DBeaver and scroll inside it

**Comment by caopuzheng** [comment]
It has not been an issue for my DBeaver until recently - I don't think anything changed except for a DBeaver upgrade.

**Comment by rafael-c-pires** [comment]
It seems to work fine when running DBeaver as an Admin, although this is not a solution for me, as I can't run an app fully as an admin on the company laptop. 

I saw the solution in a Reddit post for Eclipse, apparently with the same issue. I can run a script or software for a few minutes as an admin here, and during that time the horizontal scroll worked fine on DBeaver.

https://www.reddit.com/r/logitech/comments/y7go6n/mx_master_horizontal_scrolling_in_eclipse/

**Comment by braydenvanm** [comment]
As mentioned above, running as admin seems to work. I am having the same issue recently when not running as an admin.

**Comment by ercag** [comment]
> As mentioned above, running as admin seems to work. I am having the same issue recently when not running as an admin.

actually i didn't tried "as admin". Now i did, it worked.

**Comment by flyingcrp** [comment]
I'm encountering the same issue with master 3s and Windows 10, software version: 24.2.5.

**Comment by Hasan-J** [comment]
Encountering same issue with mx master 3s, windows 11 and dbeaver 24.2.5

**Comment by Juanse1595** [comment]
Hi, I just bought a MX Master 3S and yeah, it's still a thing. The horizontal scroll bar kind of flashes or something when trying to horizontal scroll with the horizontal whell and don't move at all. Windows 11 Dbeaver 24.3.0.

**Comment by emanueleffe** [comment]
Hi, same problem here. MX Master 2s with dBeaver 25.0.2.202504061727 on Windows 11 24H2. Unfortunately I'm unable to run it as admin.

**Comment by aydogduyunus** [comment]
+1 – Confirmed on Windows 11 with DBeaver v25.0.4 and Logitech MX Master 3S.  
The horizontal scroll bar moves visually when the DBeaver window is focused, but the table content does not scroll.  
However, when another window (like Notepad or Visio) is active and the mouse hovers over the result grid, horizontal scrolling works perfectly.

This issue clearly persists in the latest version and seems related to SWT repaint/focus handling.

**Comment by CComparon** [comment]
@aydogduyunus Spot on! I'm observing the exact same behavior with my Logitech Lift

**Comment by AtlasShruggs** [comment]
version  25.2.4.202511021717
MX Master 3S
It's working!

UPD 3 hrs later: it glitched again :(

**Comment by rafael-c-pires** [comment]
Very weird, but the related here seems to apply to Dbeaver too:

https://www.reddit.com/r/logitech/comments/1k0l1li/mx_master_3_horizontal_scroll_issue_weird/

If you rename the dbeaver.exe file to code.exe, the horizontal scroll works, even when not running as admin. 
I don't think this is a solution, and I don't know if it could cause other issues, but I'm adding it here in case it can help solve the actual problem.

**Comment by JamesG3D** [comment]
> Very weird, but the related here seems to apply to Dbeaver too:
> 
> https://www.reddit.com/r/logitech/comments/1k0l1li/mx_master_3_horizontal_scroll_issue_weird/
> 
> If you rename the dbeaver.exe file to code.exe, the horizontal scroll works, even when not running as admin. I don't think this is a solution, and I don't know if it could cause other issues, but I'm adding it here in case it can help solve the actual problem.

Could be a hack in the logitech drivers to fix horizontal scrolling in vscode? Maybe we should raise a bug with logitech instead 🤔 😅

**Comment by arnovanoordt** [comment]
> > Very weird, but the related here seems to apply to Dbeaver too:
> > https://www.reddit.com/r/logitech/comments/1k0l1li/mx_master_3_horizontal_scroll_issue_weird/
> > If you rename the dbeaver.exe file to code.exe, the horizontal scroll works, even when not running as admin. I don't think this is a solution, and I don't know if it could cause other issues, but I'm adding it here in case it can help solve the actual problem.
> 
> Could be a hack in the logitech drivers to fix horizontal scrolling in vscode? Maybe we should raise a bug with logitech instead 🤔 😅

Crazy as it sounds, this actually works 💪
It still scrolls quite slow and you'd probably need to change the name after every update, but it's a start

**Comment by mikesimons** [comment]
Not perfect but if you're using Logi Options+ you can map the thumbwheel scroll to left / right arrow keys just for dbeaver
